### PR TITLE
Added the ability to pass labels as an object to labels() and remove()

### PR DESCRIPTION
**Pull request by boris-chernysh**
Based on the issue #254 

But additionally solves another problem:
```javascript
const c = new Counter({ name: 'test', help: 'test', labelNames: ['foo', 'bar','baz'] })
c.inc({ foo: 1, baz: 2 })
c.remove(1, undefined, 2) // dont work
c.remove(1, null, 2) // dont work
c.remove({ foo: 1, baz: 2 }) // work
```
Also now calls can be visually consistent.

I don't think this is a big problem, so I didn't remove the old behavior to avoid breaking changes.

(The deleted test was a duplicate. See above)

### CODE REVIEW

**Review by SimenB — CHANGES_REQUESTED** [review]
I like it 🙂 Can you update the TS types as well?

**Review by siimon — APPROVED** [review]
No other comments than the type definitions as @SimenB mentioned 👍

**Review by SimenB — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -297,7 +297,7 @@ const gauge = new client.Gauge({
 // 1st version: Set value to 100 with "method" set to "GET" and "statusCode" to "200"
 gauge.set({ method: 'GET', statusCode: '200' }, 100);
 // 2nd version: Same effect as above
-gauge.labels('GET', '200').set(100);
```
Comment (SimenB) [inline README.md:300]: I think we should still keep this in the docs, but perhaps below the object one?

File: test/counterTest.js [hunk 2]
```diff
@@ -80,13 +90,6 @@ describe('counter', () => {
 				expect(fn).toThrowErrorMatchingSnapshot();
 			});
 
-			it('should throw error if label lengths does not match', () => {
```
Comment (SimenB) [inline test/counterTest.js:83]: it should still throw an error unless the argument is a plain error

**Review by SimenB — APPROVED** [review]

File: test/counterTest.js (hunk 2 above)
Comment (boris-chernysh) [reply test/counterTest.js:83]: I removed these lines because this test is a complete duplicate of the one above it

File: README.md (hunk 1 above)
Comment (boris-chernysh) [reply README.md:300]: yes I agree, I will do it soon

Comment (boris-chernysh) [reply README.md:310]: done

**Comment by stavalfi** [comment]
can we merge this? 👍

**Review by zbjornson — APPROVED** [review]

**Comment by zbjornson** [comment]
Landed in 3a86d05. Sorry for the delay and thanks for the contribution!

**Pull request closed without merging**